### DISCUSSION TITLE
When using routing i have to define the binding for each of the routes when i instance Hono?

### DISCUSSION THREAD

**Discussion by jonathan-gantes** [honojs/hono#2257] [Q&A]
Im checking your guide https://hono.dev/guides/best-practices#building-a-larger-application
I wan to use routing because i want to create one api for all my liteweight requests of catalogs data, my question its if i need to add the bindings in each of the app = new Hono() of each route or is there any way to inherit context/bindings to each route from main router?

and other thing if i create an env on the main router can i use it on child routes?

**Comment by yusukebe** [comment]
> When using routing i have to define the binding for each of the routes when i instance Hono? 

You don't need it. If you define the Bindings or Variables, they are used all route as the same. Creating and using instances is for routing purposes; only one instance actually handles the request.

**Comment by liuqiaowei512** [comment]
@yusukebe it's true that the bindings are used in all routes. However(might be limitation of typescript), the binding types are not correctly inferred. You can find the type mismatch in below example:

First define a binding
```typescript
/CustomBinding.ts
export type HonoBindings = {
  SOME_ENVIRONMENT_VARIABLE: string;
};
```

Use above binding for the main app
```typescript
/app.ts
const app = new Hono<{ Bindings: HonoBindings }>();
```

In a separate route, the binding of SOME_ENVIRONMENT_VARIABLE is missing
```typescript
const childRouter = new Hono()
childRouter.get("/", c => {
      console.log(c.env.SOME_ENVIRONMENT_VARIABLE) //This will give type error: 'c.env' is of type 'unknown'
      return "Hello World"
})
```

**Comment by yusukebe** [comment]
@jonathan-gantes 

Please add the type definition to the `childRouter`:

```ts
const childRouter = new Hono<{ Bindings: HonoBindings }>()
```
